src/Perl6/Metamodel/EnumHOW.nqp

```diff
@@ -70,6 +72,15 @@ class Perl6::Metamodel::EnumHOW
     method add_enum_value($obj, $value) {
         %!values{nqp::unbox_s($value.key)} := $value.value;
         @!enum_value_list[+@!enum_value_list] := $value;
+        my %value_to_enum := %!value_to_enum;
+        $lock.protect: {
+            if %value_to_enum {
+                %value_to_enum{$value.value} := $value.key;
+            }
+            nqp::scwbdisable();
+            %!value_to_enum := %value_to_enum;
+            nqp::scwbenable();
```

Comment:
And here you'd just reset the `$!value_to_enum` to `NQPMu` and let it be regenerated later.